java/mykola-melnyk/src/com/zoolatech/lecture6/tasks/_1/Task1.java

```diff
@@ -0,0 +1,78 @@
+package com.zoolatech.lecture6.tasks._1;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Class City has the following properties: string name, string country, int population.
+ * Create methods (all of them need to use Stream API):
+ * a) that accepts a list of cities and a value of a population,
+ * and returns true if all cities in the list have the population bigger than a passed value
+ * b) that accepts a list of cities and a country name and returns a city with the biggest
+ * population for that country
+ * c) that accepts a list of cities and returns a map, where the key is the city name and
+ * a value - is a corresponding city object (note: the list might contain duplicates).
+ */
+
+public class Task1 {
+
+    private static boolean checkPopulationIsBigger(ArrayList<City> cityList, int population) {
+        long populationCount = cityList.stream()
+                .filter(city -> city.getPopulation() > population)
+                .count();
+        return cityList.size() == populationCount;
+    } //Subtask a)
+
+    private static City biggestHere(ArrayList<City> cityList, String countryName) {
+        List<City> biggestCity = cityList.stream()
+                .filter(city -> city.getCountry().equalsIgnoreCase(countryName))
+                .sorted(new PopulationComparator())
+                .toList();
+        return biggestCity.get(biggestCity.size() - 1);
+    } // Subtask b) ver 1
+
+    private static City biggestHere2(ArrayList<City> cityList, String countryName) {
+        Optional<City> biggest = cityList.stream()
+                .filter(city -> city.getCountry().equalsIgnoreCase(countryName))
+                .max(new PopulationComparator());
+        return biggest.orElseThrow(() ->new IllegalArgumentException("Invalid Country " + countryName));
+    } // Subtask b) ver 2
+
+    private static Map<String, City> cityMap(ArrayList<City> cityList) {
+        return cityList.stream().collect(Collectors.toMap(City::getName, Function.identity()));
```

Comment:
your code won't be able to handle duplicate city names (see task description)

Reply:
Added Stream. distinct() method.